src/hookable.ts

```diff
@@ -218,18 +218,21 @@ export class Hookable<
         ? { name, args: arguments_, context: {} }
         : undefined;
     if (this._before) {
-      callEachWith(this._before, event);
+      callEachWith([...this._before], event);
     }
-    const result = caller(this._hooks[name] || [], arguments_);
+    const result = caller(
+      name in this._hooks ? [...this._hooks[name]] : [],
+      arguments_
+    );
     if ((result as any) instanceof Promise) {
       return result.finally(() => {
         if (this._after && event) {
-          callEachWith(this._after, event);
+          callEachWith([...this._after], event);
         }
       });
     }
     if (this._after && event) {
-      callEachWith(this._after, event);
+      callEachWith([...this._after], event);
```

Comment:
We are not having `beforeEachOne` does it makes sense to also clone before/after arrays?

Reply:
it is replicable by calling the unregister callback from `beforeEach()`

Reply:
Hmm then why not doing clone by default inside `callEachWith` if all usages have it?